packages/discovery/src/local-peer-cache/index.ts

```diff
@@ -28,6 +28,16 @@ export const DEFAULT_LOCAL_TAG_NAME = Tags.LOCAL;
 const DEFAULT_LOCAL_TAG_VALUE = 50;
 const DEFAULT_LOCAL_TAG_TTL = 100_000_000;
 
+// dynamically importing the local storage polyfill for node environments
+if (typeof window === "undefined") {
+  try {
+    const { LocalStorage } = await import("node-localstorage");
+    global.localStorage = new LocalStorage("./mock-localstorage");
```

Comment:
Is there any benefit of providing this to `NodeJS`?

To me just ignoring local storage is already good enough, better way would be to prevent it from running in `NodeJS`.

Reply:
I am concerned about us trying to add polyfills to library for the browser only - this can lead to unintentional leak of the polyfill to browser. Did you check this? 

> failing action, and an error log associated

as for this - we can do a better logging then and add getter for `localStorage` with embedded error handling

how does this sound?

Reply:
> I am concerned about us trying to add polyfills to library for the browser only 

We are introducing the polyfill for nodejs, not browsers. `node-localstorage` is a nodejs dependency that creates a mock localstorage.

> this can lead to unintentional leak of the polyfill to browser. Did you check this?

Yes - the browser tests do check for this potential leak.